core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1710,6 +1695,18 @@ class Log(@volatile var dir: File,
     }
   }
 
+  /**
+   * Get the segments in the log which have a non-empty offset intersection with the
+   * inbound swap segment
+   */
+  private def swapSegmentIntersection(swapSegment: LogSegment): Iterable[LogSegment] = {
+    lock synchronized {
+      logSegments(swapSegment.baseOffset, swapSegment.readNextOffset).filter { segment =>
```

Comment:
Can you please elaborate why this logic was changed?

Reply:
There was an off-by-one error in the original version (specifically the check on line 476) which could result in deletion of a segment outside of the swap range (I discovered this by accidentally tweaking one of the existing tests). I fixed the bug and simplified the check.